backend/src/test/java/com/htecgroup/skynest/service/impl/BucketServiceImplTest.java

```diff
@@ -0,0 +1,41 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.response.BucketResponse;
+import com.htecgroup.skynest.repository.BucketRepository;
+import com.htecgroup.skynest.utils.BucketEntityUtil;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class BucketServiceImplTest {
+
+  @Mock private BucketRepository bucketRepository;
+  @Spy private ModelMapper modelMapper;
```

Comment:
In every test class we have it's annotated with Spy, should we change it ?

Reply:
the ModelMapper ??

Reply:
yes the ModelMapper in the tests classes.

Reply:
well yes

Reply:
If it's not Spy, then it throws null pointer exception, if we mock it with Mock then we need to use Model mapper in almost every test, so this was the idea by @MetoMladenovski for excluding Model mapper from every test refactor.

Reply:
Didn't Meto already exclude that for the tests, and use the "mapped" object instead?

Reply:
I am not sure I understand, but I did leave it as a @Spy ModelMapper, because we use it too much in our service and all it does is map the Entity to Response,Dto etc. 
If I mock the ModelMapper we would have to mock all those .map() methods in the services. My logic was to use @Spy to not get the NullPointerException, and have modelMapper map those objects.